test/test_transforms.py

```diff
@@ -1794,6 +1549,292 @@ def test_randomness(fn, trans, config, p):
     assert p_value > 0.0001
 
 
+@pytest.mark.parametrize('data_type, mode', [
+    ('float', 'L'),
+    ('byte', 'L'),
+    ('short', 'I;16'),
+    ('int', 'I'),
+])
+def test_1_channel_tensor_to_pil_image(data_type, mode):
+    to_tensor = transforms.ToTensor()
+
+    if data_type == 'float':
+        img_data = torch.Tensor(1, 4, 4).uniform_()
+        expected_output = img_data.mul(255).int().float().div(255).numpy()
+    elif data_type == 'byte':
+        img_data = torch.ByteTensor(1, 4, 4).random_(0, 255)
+        expected_output = img_data.float().div(255.0).numpy()
+    elif data_type == 'short':
+        img_data = torch.ShortTensor(1, 4, 4).random_()
+        expected_output = img_data.numpy()
+    elif data_type == 'int':
+        img_data = torch.IntTensor(1, 4, 4).random_()
+        expected_output = img_data.numpy()
+    else:
+        assert False
```

Comment:
Instead of this, we could instead parametrize over tuples (this will replace the `zip`, basically):

```py
@pytest.mark.parametrize('img_data, expected_output, expected_mode', [
    (torch.Tensor(1, 4, 4).uniform_(), img_data_float.mul(255).int().float().div(255).numpy(), 'L'),
    ....
])
 def test_1_channel_tensor_to_pil_image(img_data, expected_output, expected_mode):
```

Also I think we should keep the naming `expected_mode` instead of `mode`

Reply:
Hi @NicolasHug, I'm not sure I understand accurately your suggestion here and please point out my misunderstanding.

Because `img_data` is generated randomly, the `expected_output` also depends on `img_data`. If we change it to

```python
(torch.Tensor(1, 4, 4).uniform_(), torch.Tensor(1, 4, 4).uniform_().mul(255).int().float().div(255).numpy(), 'L'),
...
```

this above two value will be inconsistent.

Reply:
ah yes, sorry I missed that. In this case we can still parametrize over `'img_data, expected_output, expected_mode'` as I suggested, but maybe we'll need to generate the data into a helper like

```py
_get_input_data():   # find a better name for that :)
    img_data_float = torch.Tensor(1, 4, 4).uniform_()
    expected_output =  img_data_float.mul(255).int().float().div(255).numpy()
    yield img_data_float, expected_output, 'L'
    # ...
    yield ...
    yield ...

# and then
@pytest.mark.parametrize('img_data, expected_output, expected_mode', _get_input_data())
 def test_1_channel_tensor_to_pil_image(img_data, expected_output, expected_mode):
    ...
```

Reply:
Done as suggested, and that's the first time I understand how `yield` works :)